src/lib/ERC4337/validator/TurnkeyValidator.sol

```diff
@@ -0,0 +1,60 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.13;
+
+import {Validator} from "src/lib/ERC4337/validator/Validator.sol";
+import {UserOperation} from "src/lib/ERC4337/utils/UserOperation.sol";
+import {Operations} from "src/lib/Operations.sol";
+import {Access} from "src/access/Access.sol";
+import {SignatureChecker} from "openzeppelin-contracts/utils/cryptography/SignatureChecker.sol";
+
+/// @dev Validator module that restricts valid signatures to only come from recognized Turnkeys
+/// for the calling Accounts contract 
+contract TurnkeyValidator is Validator {
+
+    constructor(address _entryPointAddress) Validator(_entryPointAddress) {}
+
+    /// @dev This example contract would only be forwarded signatures formatted as follows:
+    /// `abi.encodePacked(address signer, bytes memory eoaSig)`
+    function validateUserOp(
+        UserOperation calldata userOp, 
+        bytes32 userOpHash, 
+        uint256 missingAccountFunds
+    ) external virtual returns (uint256 validationData) {
+        // silence compiler by discarding unused variable
+        (missingAccountFunds);
+
+        /// @notice BLS sig aggregator and timestamp expiry are not currently supported by this contract 
+        /// so `bytes20(0x0)` and `bytes6(0x0)` suffice. To enable support for aggregator and timestamp expiry,
+        /// override the following params
+        bytes20 authorizer;
+        bytes6 validUntil;
+        bytes6 validAfter;
+        uint256 successData = uint256(bytes32(abi.encodePacked(authorizer, validUntil, validAfter)));
+
+        // deconstruct signature into `(validator, nestedSignature)`
+        (address signer, bytes memory nestedSignature) = abi.decode(userOp.signature, (address, bytes));

```

Comment:
I'm thinking that `abi.encode`/`abi.decode` may cause accidental bugs for us given they encode a length of variable-length data. I think we can just manually parse the first 20 bytes as the signer and the remaining bytes as a signature to validate the signer's intent.